src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,85 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFALT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V> [] buckets;
+    private double threshold;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFALT_CAPACITY];
+        threshold = DEFALT_CAPACITY * LOAD_FACTOR;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int index = getBucket(key, buckets.length);
+        Node<K, V> element = findNode(key, index);
+        if (element != null) {
+            element.value = value;
+        } else {
+            Node<K, V> entry = new Node<>(key, value, buckets[index]);
+            buckets[index] = entry;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> element = findNode(key, getBucket(key, buckets.length));
+        return element != null ? element.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getBucket(K key, int capacity) {
```

Comment:
```suggestion
    private int getBucketIndex(K key) {
```
you get the index, not the bucket, do you? by the way, this method always uses `buckets.length`, so I think, there is no need to use a `capacity` parameter